sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -0,0 +1,394 @@
+        /// </summary>
+        ///
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            ActiveConnection?.Dispose();
+            _disposed = true;
+        }
+
+        /// <summary>
+        ///   Creates an AMQP connection for a given scope.
+        /// </summary>
+        ///
+        /// <param name="amqpVersion">The version of AMQP to use for the connection.</param>
+        /// <param name="serviceEndpoint">The endpoint for the Event Hubs service to which the scope is associated.</param>
+        /// <param name="transportType">The type of transport to use for communication.</param>
+        /// <param name="proxy">The proxy, if any, to use for communication.</param>
+        /// <param name="scopeIdentifier">The unique identifier for the associated scope.</param>
+        /// <param name="timeout">The timeout to consider when creating the connection.</param>
+        ///
+        /// <returns>An AMQP connection that may be used for communicating with the Event Hubs service.</returns>
+        ///
+        protected virtual async Task<AmqpConnection> CreateConnectionAsync(Version amqpVersion,
+                                                                           Uri serviceEndpoint,
+                                                                           TransportType transportType,
+                                                                           IWebProxy proxy,
+                                                                           string scopeIdentifier,
+                                                                           TimeSpan timeout)
+        {
+            var hostName = serviceEndpoint.Host;
+            var amqpSettings = CreateAmpqSettings(AmqpVersion);
+            var connectionSetings = CreateAmqpConnectionSettings(hostName, scopeIdentifier);
+
+            var transportSettings = transportType.IsWebSocketTransport()
+                ? CreateTransportSettingsForWebSockets(hostName, proxy)
+                : CreateTransportSettingsforTcp(hostName, serviceEndpoint.Port);
+
+            // Create and open the connection, respecting the timeout constraint
+            // that was received.
+
+            var stopWatch = Stopwatch.StartNew();
+
+            var initiator = new AmqpTransportInitiator(amqpSettings, transportSettings);
+            var transport = await initiator.ConnectTaskAsync(timeout).ConfigureAwait(false);
+
+            var connection = new AmqpConnection(transport, amqpSettings, connectionSetings);
+            await connection.OpenAsync(timeout.CalculateRemaining(stopWatch.Elapsed)).ConfigureAwait(false);
+
+            stopWatch.Stop();
+
+            // Create the CBS link that will be used for authorization and ensure that it is associated
+            // with the connection.
+
+            var cbsLink = new AmqpCbsLink(connection);
+
+            // TODO (pri2 // squire):
```

Comment:
Thus far, this is never hit.   I want to keep in in place until I finish the producer/consumer work, just in case.